integration-tests/subscriptions.test.ts

```diff
@@ -65,27 +74,146 @@ if (SUBSCRIPTIONS_ENABLED === 'true') {
                 const subResource = randomSubscription(uuid);
                 const postResult = await client.post('Subscription', subResource);
                 expect(postResult.status).toEqual(201);
-                const resourceThatMatchesSubscription = {
-                    resourceType: 'Patient',
-                    name: [
-                        {
-                            given: ['Smith'],
-                            family: 'Smith',
-                        },
-                    ],
-                };
-                // post matching resource on another tenant
+                // post matching resource on another tenant, waiting 1 minute for subscription to enable
+                await sleep(60_000);
                 const postPatientResult = await clientAnotherTenant.post('Patient', resourceThatMatchesSubscription);
                 expect(postPatientResult.status).toEqual(201);
-                // give 2 minutes for notification to be placed in ddb table
-                await new Promise((r) => setTimeout(r, 120_000));
+                // give 1 minute for notification to be placed in ddb table
+                await sleep(60_000);
                 // make sure no notification was receieved for first tenant
                 const notifications = await subscriptionsHelper.getNotifications(
                     `${uuid}/Patient/${postPatientResult.data.id}`,
                 );
                 expect(notifications).toEqual([]);
             });
         }
+
+        test('invalid subscriptions', async () => {
+            // test for endpoints that aren't allow listed
+            let subResource = randomSubscription(v4());
+            subResource.channel.endpoint = 'https://non_allow_listed_endpoint.com';
+            await expect(client.post('Subscription', subResource)).rejects.toMatchObject({
+                response: { status: 400 },
+            });
+
+            // test for unsupported channel types (email, sms, and websocket)
+            subResource = randomSubscription(v4());
+            subResource.channel.type = 'email';
+            await expect(client.post('Subscription', subResource)).rejects.toMatchObject({
+                response: { status: 400 },
+            });
+            subResource.channel.type = 'SMS';
```

Comment:
this needs to be `sms` (case sensitive). Make sure that you are using the codes as they appear on https://fhir-ru.github.io/valueset-subscription-channel-type.html#expansion across this tests

Here you are getting a 404 because 'SMS' is an invalid code, so the validator (either HAPI or JSON schema) reject it, but what we want to test is the custom logic we wrote to reject valid channels that we don't support

Reply:
Gotcha, I've updated the test to the correct spelling and also updated websocket to match the standard as well.